airbyte-integrations/bases/standard-test/standard_test/tests/test_incremental.py

```diff
@@ -22,8 +22,35 @@
 SOFTWARE.
 """
 
+from airbyte_protocol import Type
+
 from standard_test import BaseTest
+from standard_test.connector_runner import ConnectorRunner
+from standard_test.utils import filter_output, incremental_only_catalog
 
 
 class TestIncremental(BaseTest):
-    pass
+    def test_sequential_reads(self, connector_config, configured_catalog, docker_runner: ConnectorRunner):
+        configured_catalog = incremental_only_catalog(configured_catalog)
+        output = docker_runner.call_read(connector_config, configured_catalog)
+
+        records_1 = filter_output(output, type_=Type.RECORD)
+        states_1 = filter_output(output, type_=Type.STATE)
+
+        assert states_1, "Should produce at least one state"
+        assert records_1, "Should produce at least one record"
+
+        latest_state = states_1[-1].data
+        output = docker_runner.call_read_with_state(connector_config, configured_catalog, state={})
+        records_2 = filter_output(output, type_=Type.RECORD)
+        states_2 = filter_output(output, type_=Type.STATE)
+
+        assert states_1 == states_2, "Empty state should has same result as no state"
+        assert records_1 == records_2, "Empty state should has same result as no state"
+
+        output = docker_runner.call_read_with_state(connector_config, configured_catalog, state=latest_state)
+        records_3 = filter_output(output, type_=Type.RECORD)
+        states_3 = filter_output(output, type_=Type.STATE)
+
+        assert not records_3, "There should be no records after second incremental read"
```

Comment:
well, there should be better check here

Reply:
Ideally this verifies the incremental sync invariant described [here](https://docs.google.com/document/d/1pFLR8eLP5VOj6vDtWe69KZL0aWWHEA6Vkbod6NYJon4/edit#)